src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -2026,6 +2108,12 @@ ColumnPtr executeStringInteger(const ColumnsWithTypeAndName & arguments, const A
                 left_col = castColumn(arguments[0], converted_type);
                 right_col = castColumn(arguments[1], converted_type);
             }
+            else if constexpr (is_div_int_with_decimal)
+            {
+                const auto converted_type = std::make_shared<DataTypeInt64>();
+                left_col = castColumn(arguments[0], converted_type);
+                right_col = castColumn(arguments[1], converted_type);
+            }
```

Comment:
So, we actually don't do real decimal division, we just cast both arguments to `Int64` and then devide them as integers. But it won't work, we will lose precision, and we cannot convert for examplt `Decimal128` to `Int64`. Examples:
```
avogar-dev :) select intDiv(22.8::Decimal32(2), 0.1::Decimal32(2))

Received exception:
Code: 153. DB::Exception: Division by zero: While processing intDiv(CAST('22.8', 'Decimal32(2)'), CAST('0.1', 'Decimal32(2)')). (ILLEGAL_DIVISION)

avogar-dev :) select intDiv(222222222222222222222222222222222222.8::Decimal128(2), 2::Decimal32(2))

Received exception:
Code: 407. DB::Exception: Convert overflow: While processing intDiv(CAST('222222222222222222222222222222222222.8', 'Decimal128(2)'), CAST('2', 'Decimal32(2)')). (DECIMAL_OVERFLOW)
```

I think we should do real decimal division and after it cast it to result integer. Need to think how to do it properly

Reply:
I considered this problem and didn't know how to resolve it as well, so came up with this solution, though it can be not good